float64 distance
float64 velocity
int32  direction
---
bool done